src/Features/VisualBasic/Portable/CodeFixes/GenerateType/GenerateTypeCodeFixProvider.vb

```diff
@@ -56,7 +57,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeFixes.GenerateType
         End Function
 
         Protected Overrides Function GetTargetNode(node As SyntaxNode) As SyntaxNode
-            Return (DirectCast(node, ExpressionSyntax)).GetRightmostName()
+            Dim qualified = TryCast(node.Parent, QualifiedNameSyntax)
+            If qualified IsNot Nothing Then
+                Return GetRightMostName(qualified)
+            End If
+
+            Dim memberAccess = TryCast(node.Parent, MemberAccessExpressionSyntax)
+            If memberAccess IsNot Nothing Then
+                Return GetRightMostName(memberAccess)
+            End If
+
+            Return DirectCast(node, ExpressionSyntax).GetRightmostName
+        End Function
+
+        Private Shared Function GetRightMostName(Of TNameSyntax As ExpressionSyntax)(ByRef expression As TNameSyntax) As SyntaxNode
```

Comment:
Why is this parameter ByRef? The value it gets changed to is never observed by the calling code.